dht/token.hh

```diff
@@ -65,12 +68,20 @@ public:
     //     [0x80] == 0.5
     //     [0x00, 0x80] == 1/512
     //     [0xff, 0x80] == 1 - 1/512
-    managed_bytes _data;
+    std::array<uint8_t, 8> _data;
 
     token() : _kind(kind::before_all_keys) {
     }
 
-    token(kind k, managed_bytes d) : _kind(std::move(k)), _data(std::move(d)) {
+    token(kind k, std::array<uint8_t, 8> d) : _kind(std::move(k)), _data(std::move(d)) {
+    }
+
+    token(kind k, const bytes& b) : _kind(std::move(k)) {
+        std::copy_n(b.begin(), 8, _data.begin());
```

Comment:
Can we assume 8 bytes here? Can we get more from an old node that may support different kind of tokens. Or less?

Reply:
We assume only Murmur3 is in use on the cluster and it always uses 8 bytes to represent tokens. Support for partitioners other than Murmur3 was dropped here: https://github.com/scylladb/scylla/pull/5636

Reply:
Assumptions are good, but don't we want to verify that they are true?

Reply:
Sure. I will add a check. Thanks.